Problem working with windows.

Hi,

I tried using this library with Windows 10 as an administrator and faced the same issues as discussed by people earlier. When I restricted my privileges everything seems to work fine.  

Although, the changes made by after https://github.com/yandex-qatools/postgresql-embedded/pull/29 merge request worked fine but future commits to the library reverted this change. 

I am not willing to change my machine privileges but my project is highly dependent on this library.
Can you help me with the solution?

For your ref:
I am getting same logs as in https://github.com/yandex-qatools/postgresql-embedded/issues/28

I used the below code to test on my Windows 7 machine. 

```
package Test.unit;

import de.flapdoodle.embed.process.config.IRuntimeConfig;
import de.flapdoodle.embed.process.distribution.Distribution;
import de.flapdoodle.embed.process.io.directories.FixedPath;
import de.flapdoodle.embed.process.store.PostgresArtifactStoreBuilder;
import org.junit.Test;
import ru.yandex.qatools.embed.postgresql.Command;
import ru.yandex.qatools.embed.postgresql.EmbeddedPostgres;
import ru.yandex.qatools.embed.postgresql.PackagePaths;
import ru.yandex.qatools.embed.postgresql.config.PostgresDownloadConfigBuilder;
import ru.yandex.qatools.embed.postgresql.config.RuntimeConfigBuilder;

import java.io.IOException;
import java.util.List;

import static java.util.Arrays.asList;
import static org.apache.commons.lang3.StringUtils.join;

public class TestBlah {

    @Test
    public void blah() throws IOException {
        final Command cmd = Command.Postgres;
        final FixedPath cachedDir = new FixedPath("C:\\a-tmp");
        IRuntimeConfig build = new RuntimeConfigBuilder()
                .defaults(cmd)
                .commandLinePostProcessor((Distribution distribution, List<String> args) ->
                {
                    return asList("runas /trustlevel:0x20000", join(args, " "));
                })
                .artifactStore(new PostgresArtifactStoreBuilder()
                        .defaults(cmd)
                        .tempDir(cachedDir)
                        .download(new PostgresDownloadConfigBuilder()
                                .defaultsForCommand(cmd)
                                .packageResolver(new PackagePaths(cmd, cachedDir))
                                .build()))
                .build();

        new EmbeddedPostgres().start(build);

    }
}
```

And this was the reply my system gave me.

```
Extract C:\Users\Jewel\.embedpostgresql\postgresql-10.1-2-windows-x64-binaries.zip START
...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................Extract C:\Users\Jewel\.embedpostgresql\postgresql-10.1-2-windows-x64-binaries.zip DONE
23:28:51.494 [main] ERROR d.f.e.p.runtime.AbstractProcess - failed to call start() 
java.io.IOException: Cannot run program "runas /trustlevel:0x20000": CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at de.flapdoodle.embed.process.runtime.ProcessControl.start(ProcessControl.java:205)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:98)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.InitDbProcess.<init>(InitDbProcess.java:56)
	at ru.yandex.qatools.embed.postgresql.InitDbExecutable.start(InitDbExecutable.java:23)
	at ru.yandex.qatools.embed.postgresql.InitDbExecutable.start(InitDbExecutable.java:14)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.runCmd(PostgresProcess.java:119)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.runCmd(PostgresProcess.java:74)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.onBeforeProcess(PostgresProcess.java:196)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:82)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.<init>(PostgresProcess.java:68)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:25)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:13)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:140)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:113)
	at Test.unit.TestBlah.blah(TestBlah.java:41)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 41 common frames omitted
23:28:51.497 [main] INFO  d.f.e.p.runtime.AbstractProcess - construct AbstractPostgresConfig{storage=Storage{dbDir=C:\Users\Jewel\AppData\Local\Temp\postgresql-embed-b7766a39-c1be-41ad-a2e4-b935d8bd17ff\db-content-e1e46f6a-e74a-458c-840f-4429759fe90e, dbName='postgres', isTmpDir=true}, network=Net{host='localhost', port=64517}, timeout=Timeout{startupTimeout=15000}, credentials=Credentials{username='postgres', password='postgres'}, args=[], additionalInitDbParams=[-E, SQL_ASCII, --locale=C, --lc-collate=C, --lc-ctype=C]} 
23:28:51.502 [main] WARN  d.f.e.p.runtime.AbstractProcess - Could not delete pid file: C:\a-tmp\pgsql-10.1-1\pgsql\bin\initdb.pid 
23:28:51.502 [main] WARN  r.y.q.e.postgresql.PostgresProcess - Failed to run command initdb 
java.io.IOException: Cannot run program "runas /trustlevel:0x20000": CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at de.flapdoodle.embed.process.runtime.ProcessControl.start(ProcessControl.java:205)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:98)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.InitDbProcess.<init>(InitDbProcess.java:56)
	at ru.yandex.qatools.embed.postgresql.InitDbExecutable.start(InitDbExecutable.java:23)
	at ru.yandex.qatools.embed.postgresql.InitDbExecutable.start(InitDbExecutable.java:14)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.runCmd(PostgresProcess.java:119)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.runCmd(PostgresProcess.java:74)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.onBeforeProcess(PostgresProcess.java:196)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:82)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.<init>(PostgresProcess.java:68)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:25)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:13)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:140)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:113)
	at Test.unit.TestBlah.blah(TestBlah.java:41)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 41 common frames omitted
23:28:51.504 [main] ERROR d.f.e.p.runtime.AbstractProcess - failed to call start() 
java.io.IOException: Cannot run program "runas /trustlevel:0x20000": CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at de.flapdoodle.embed.process.runtime.ProcessControl.start(ProcessControl.java:205)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:98)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.<init>(PostgresProcess.java:68)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:25)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:13)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:140)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:113)
	at Test.unit.TestBlah.blah(TestBlah.java:41)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 32 common frames omitted
23:28:51.504 [main] INFO  d.f.e.p.runtime.AbstractProcess - construct AbstractPostgresConfig{storage=Storage{dbDir=C:\Users\Jewel\AppData\Local\Temp\postgresql-embed-b7766a39-c1be-41ad-a2e4-b935d8bd17ff\db-content-e1e46f6a-e74a-458c-840f-4429759fe90e, dbName='postgres', isTmpDir=true}, network=Net{host='localhost', port=64517}, timeout=Timeout{startupTimeout=15000}, credentials=Credentials{username='postgres', password='postgres'}, args=[], additionalInitDbParams=[-E, SQL_ASCII, --locale=C, --lc-collate=C, --lc-ctype=C]} 

java.lang.NullPointerException
	at de.flapdoodle.embed.process.runtime.AbstractProcess.getProcessId(AbstractProcess.java:232)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.isProcessRunning(AbstractProcess.java:225)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.stopInternal(PostgresProcess.java:145)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.stop(AbstractProcess.java:170)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:119)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.<init>(PostgresProcess.java:68)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:25)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:13)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:140)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:113)
	at Test.unit.TestBlah.blah(TestBlah.java:41)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
```

Made the change.
```
                .defaults(cmd)
                .commandLinePostProcessor((distribution, args) -> asList("runas /trustlevel:0x20000",  join(Stream.concat(Stream.of("cmd", "/c"), args.stream()).collect(Collectors.toList()), " ")))
                .artifactStore(new PostgresArtifactStoreBuilder()
```

Here's the output.

```
"C:\Program Files\Java\jdk1.8.0_152\bin\java" -ea -Didea.test.cyclic.buffer.size=1048576 "-javaagent:C:\Program Files\JetBrains\IntelliJ IDEA 2017.3\lib\idea_rt.jar=49572:C:\Program Files\JetBrains\IntelliJ IDEA 2017.3\bin" -Dfile.encoding=UTF-8 -classpath "C:\Program Files\JetBrains\IntelliJ IDEA 2017.3\lib\idea_rt.jar;C:\Program Files\JetBrains\IntelliJ IDEA 2017.3\plugins\junit\lib\junit-rt.jar;C:\Program Files\JetBrains\IntelliJ IDEA 2017.3\plugins\junit\lib\junit5-rt.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\charsets.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\deploy.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\access-bridge-64.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\cldrdata.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\dnsns.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\jaccess.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\jfxrt.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\localedata.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\nashorn.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\sunec.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\sunjce_provider.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\sunmscapi.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\sunpkcs11.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\ext\zipfs.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\javaws.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\jce.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\jfr.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\jfxswt.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\jsse.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\management-agent.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\plugin.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\resources.jar;C:\Program Files\Java\jdk1.8.0_152\jre\lib\rt.jar;C:\repos\sasquatch-screening\out\test\classes;C:\repos\sasquatch-screening\out\production\classes;C:\repos\sasquatch-screening\out\production\resources;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\ru.yandex.qatools.embed\postgresql-embedded\2.6\480f5cdd26f49ec34f59693dd6185fd444076f4f\postgresql-embedded-2.6.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.projectlombok\lombok\1.16.18\557d13dcb647038dc687390711ccb5c9b3ffbd60\lombok-1.16.18.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-context-support\5.0.2.RELEASE\f3dcba6050b1635aacc606243decc99515d1430\spring-context-support-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.reflections\reflections\0.9.9\296d8adb2f22a38025f44b45cac89835ff0bbaf\reflections-0.9.9.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\javax.mail\mail\1.4.1\8b7bc69010655425dabf091b51d1e90b4de36715\mail-1.4.1.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-context-indexer\5.0.2.RELEASE\d5d4319f8cc18683126099d65face7533a002de\spring-context-indexer-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-webmvc\5.0.2.RELEASE\8dbe4a28cae3ffa222aeff6776a3928a88d66e6f\spring-webmvc-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-orm\5.0.2.RELEASE\59f207a8426f6090734663afcd6d6003cca57ce8\spring-orm-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-aspects\5.0.2.RELEASE\b26a3ab00f741144c4e2dab39c0ab0ea7a411ef\spring-aspects-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.aspectj\aspectjrt\1.8.13\bfc5a877fd80648e5467239d370735654aaf7a92\aspectjrt-1.8.13.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.aspectj\aspectjweaver\1.8.13\ad94df2a28d658a40dc27bbaff6a1ce5fbf04e9b\aspectjweaver-1.8.13.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework.security\spring-security-web\5.0.0.RELEASE\fea662f3947fdff7221d9efaa82881a671c0ca35\spring-security-web-5.0.0.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework.security\spring-security-config\5.0.0.RELEASE\64e01e68d9755d5d68076a1e7f6e79b2940a5e26\spring-security-config-5.0.0.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework.security\spring-security-core\5.0.0.RELEASE\3e3a3a68fc458dcd723216302eda7d772558506\spring-security-core-5.0.0.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-context\5.0.2.RELEASE\743a5f6d94a8af5759e6e70c360a7b4a47a704c\spring-context-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-aop\5.0.2.RELEASE\4d49aa2cff33be79da99413e28a905898600ccaa\spring-aop-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.hibernate\hibernate-core\5.2.12.Final\ea479b6692e22c9b642dd4142db670e92d106544\hibernate-core-5.2.12.Final.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.postgresql\postgresql\42.1.4\1c7788d16b67d51f2f38ae99e474ece968bf715a\postgresql-42.1.4.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.apache.commons\commons-dbcp2\2.1.1\c4f4a76171671ccf293be8995a498eab7fa8ed24\commons-dbcp2-2.1.1.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.slf4j\jcl-over-slf4j\1.7.25\f8c32b13ff142a513eeb5b6330b1588dcb2c0461\jcl-over-slf4j-1.7.25.jar;C:\Users\Jewel\.m2\repository\ch\qos\logback\logback-classic\1.2.3\logback-classic-1.2.3.jar;C:\Users\Jewel\.m2\repository\de\flapdoodle\embed\de.flapdoodle.embed.process\2.0.1\de.flapdoodle.embed.process-2.0.1.jar;C:\Users\Jewel\.m2\repository\org\slf4j\slf4j-api\1.7.25\slf4j-api-1.7.25.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.apache.commons\commons-collections4\4.1\a4cf4688fe1c7e3a63aa636cc96d013af537768e\commons-collections4-4.1.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\com.fasterxml.jackson.core\jackson-databind\2.9.2\1d8d8cb7cf26920ba57fb61fa56da88cc123b21f\jackson-databind-2.9.2.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\com.fasterxml.jackson.core\jackson-core\2.9.2\aed20e50152a2f19adc1995c8d8f307c7efa414d\jackson-core-2.9.2.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\com.fasterxml.jackson.core\jackson-annotations\2.9.2\bb0f1a93c122723906420b7a77b6c4b669da0fc2\jackson-annotations-2.9.2.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\javax.servlet\javax.servlet-api\4.0.0\60200affc2fe0165136ed3690faf00b66aed581a\javax.servlet-api-4.0.0.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-web\5.0.2.RELEASE\eea8edef792fb93c1c55cc61078ce14ffb3542df\spring-web-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-jdbc\5.0.2.RELEASE\5efc255d4073cfa639ecf8e92f162528519f9dd3\spring-jdbc-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-tx\5.0.2.RELEASE\38ef30ad6e34b5004c8955685b6a7ea753717b7d\spring-tx-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-beans\5.0.2.RELEASE\301ee07b390bc8b5691f4206411b49beb06f7ff2\spring-beans-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-expression\5.0.2.RELEASE\246bf50b5b46379041d333b4a46a01a7aea0b788\spring-expression-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-core\5.0.2.RELEASE\45b2958ab3fb022dd29f8b1c553ebf1c75a144aa\spring-core-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\com.google.guava\guava\15.0\ed727a8d9f247e2050281cb083f1c77b09dcb5cd\guava-15.0.jar;C:\Users\Jewel\.m2\repository\org\javassist\javassist\3.20.0-GA\javassist-3.20.0-GA.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\com.google.code.findbugs\annotations\2.0.1\9ef6656259841cebfb9fb0697bb122ada4485498\annotations-2.0.1.jar;C:\Users\Jewel\.m2\repository\javax\activation\activation\1.1\activation-1.1.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.hibernate.common\hibernate-commons-annotations\5.0.1.Final\71e1cff3fcb20d3b3af4f3363c3ddb24d33c6879\hibernate-commons-annotations-5.0.1.Final.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.jboss.logging\jboss-logging\3.3.0.Final\3616bb87707910296e2c195dc016287080bba5af\jboss-logging-3.3.0.Final.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.hibernate.javax.persistence\hibernate-jpa-2.1-api\1.0.0.Final\5e731d961297e5a07290bfaf3db1fbc8bbbf405a\hibernate-jpa-2.1-api-1.0.0.Final.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\antlr\antlr\2.7.7\83cd2cd674a217ade95a4bb83a8a14f351f48bd0\antlr-2.7.7.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.jboss.spec.javax.transaction\jboss-transaction-api_1.2_spec\1.0.1.Final\4441f144a2a1f46ed48fcc6b476a4b6295e6d524\jboss-transaction-api_1.2_spec-1.0.1.Final.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.jboss\jandex\2.0.3.Final\bfc4d6257dbff7a33a357f0de116be6ff951d849\jandex-2.0.3.Final.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\com.fasterxml\classmate\1.3.0\183407ff982e9375f1a1c4a51ed0a9307c598fc7\classmate-1.3.0.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\dom4j\dom4j\1.6.1\5d3ccc056b6f056dbf0dddfdf43894b9065a8f94\dom4j-1.6.1.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.apache.commons\commons-pool2\2.4.2\e5f4f28f19d57716fbc3989d7a357ebf1e454fea\commons-pool2-2.4.2.jar;C:\Users\Jewel\.m2\repository\ch\qos\logback\logback-core\1.2.3\logback-core-1.2.3.jar;C:\Users\Jewel\.m2\repository\commons-io\commons-io\2.4\commons-io-2.4.jar;C:\Users\Jewel\.m2\repository\org\apache\commons\commons-lang3\3.1\commons-lang3-3.1.jar;C:\Users\Jewel\.m2\repository\net\java\dev\jna\jna-platform\4.0.0\jna-platform-4.0.0.jar;C:\Users\Jewel\.m2\repository\net\java\dev\jna\jna\4.0.0\jna-4.0.0.jar;C:\Users\Jewel\.m2\repository\org\apache\commons\commons-compress\1.10\commons-compress-1.10.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.springframework\spring-jcl\5.0.2.RELEASE\212a0bdd54e026641cf527aeb8e578f86e402bf1\spring-jcl-5.0.2.RELEASE.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.subethamail\subethasmtp\3.1.7\41381242daa11113c32da5b25d001bfdf4df53fb\subethasmtp-3.1.7.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\javax.mail\mail\1.4.4\b907ef0a02ff6e809392b1e7149198497fcc8e49\mail-1.4.4.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\io.rest-assured\rest-assured\3.0.6\7048c659a0f1b8adca66af281edbf1397dddd63d\rest-assured-3.0.6.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\com.github.stefanbirkner\system-rules\1.16.0\1bec4d14816e23a1715d5c55db36919b5a529a4f\system-rules-1.16.0.jar;C:\Users\Jewel\.m2\repository\junit\junit\4.12\junit-4.12.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.assertj\assertj-core\3.8.0\b209d90ff2e279bee3e02547ee7b11349c52d0e3\assertj-core-3.8.0.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.mockito\mockito-core\2.12.0\a42fb82aa1eccb27c3f0371c8bc7385ae1c6598d\mockito-core-2.12.0.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\javax.activation\activation\1.1.1\485de3a253e23f645037828c07f1d7f1af40763a\activation-1.1.1.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\io.rest-assured\xml-path\3.0.6\5ea3318626ac935bfc5088c772ea6d57c43e1bf3\xml-path-3.0.6.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.codehaus.groovy\groovy-xml\2.4.12\b5c023bbbc8a4bbf9c0b38d731fe31d53f00c64a\groovy-xml-2.4.12.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\io.rest-assured\json-path\3.0.6\46b9832bdcf3b009ef29dad04e5bf1b951c2966e\json-path-3.0.6.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.codehaus.groovy\groovy-json\2.4.12\810b46f2b359b904c4c7428ad4c03c09094a01ff\groovy-json-2.4.12.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\io.rest-assured\rest-assured-common\3.0.6\fa0cb69682ed3df571cab6d20894c4ed2970b901\rest-assured-common-3.0.6.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.codehaus.groovy\groovy\2.4.12\a43be367110c491787219f1c128b5b5fc54f1e70\groovy-2.4.12.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.apache.httpcomponents\httpmime\4.5.1\96823b9421ebb9f490dec837d9f96134e864e3a7\httpmime-4.5.1.jar;C:\Users\Jewel\.m2\repository\org\apache\httpcomponents\httpclient\4.5.3\httpclient-4.5.3.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.hamcrest\hamcrest-library\1.3\4785a3c21320980282f9f33d0d1264a69040538f\hamcrest-library-1.3.jar;C:\Users\Jewel\.m2\repository\org\hamcrest\hamcrest-core\1.3\hamcrest-core-1.3.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.ccil.cowan.tagsoup\tagsoup\1.2.1\5584627487e984c03456266d3f8802eb85a9ce97\tagsoup-1.2.1.jar;C:\Users\Jewel\.m2\repository\com\google\code\findbugs\jsr305\1.3.9\jsr305-1.3.9.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\net.bytebuddy\byte-buddy\1.7.9\51218a01a882c04d0aba8c028179cce488bbcb58\byte-buddy-1.7.9.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\net.bytebuddy\byte-buddy-agent\1.7.9\a6c65f9da7f467ee1f02ff2841ffd3155aee2fc9\byte-buddy-agent-1.7.9.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\org.objenesis\objenesis\2.6\639033469776fd37c08358c6b92a4761feb2af4b\objenesis-2.6.jar;C:\Users\Jewel\.m2\repository\org\apache\commons\commons-lang3\3.4\commons-lang3-3.4.jar;C:\Users\Jewel\.m2\repository\org\apache\httpcomponents\httpcore\4.4.6\httpcore-4.4.6.jar;C:\Users\Jewel\.m2\repository\commons-codec\commons-codec\1.9\commons-codec-1.9.jar;C:\Users\Jewel\.gradle\caches\modules-2\files-2.1\javax.xml.bind\jaxb-api\2.2.12\4c83805595b15acf41d71d49e3add7c0e85baaed\jaxb-api-2.2.12.jar" com.intellij.rt.execution.junit.JUnitStarter -ideVersion5 -junit4 Test.unit.TestBlah,blah
Extract C:\Users\Jewel\.embedpostgresql\postgresql-10.1-2-windows-x64-binaries.zip START
...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................Extract C:\Users\Jewel\.embedpostgresql\postgresql-10.1-2-windows-x64-binaries.zip DONE
00:03:10.674 [main] ERROR d.f.e.p.runtime.AbstractProcess - failed to call start() 
java.io.IOException: Cannot run program "runas /trustlevel:0x20000": CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at de.flapdoodle.embed.process.runtime.ProcessControl.start(ProcessControl.java:205)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:98)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.InitDbProcess.<init>(InitDbProcess.java:56)
	at ru.yandex.qatools.embed.postgresql.InitDbExecutable.start(InitDbExecutable.java:23)
	at ru.yandex.qatools.embed.postgresql.InitDbExecutable.start(InitDbExecutable.java:14)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.runCmd(PostgresProcess.java:119)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.runCmd(PostgresProcess.java:74)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.onBeforeProcess(PostgresProcess.java:196)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:82)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.<init>(PostgresProcess.java:68)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:25)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:13)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:140)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:113)
	at Test.unit.TestBlah.blah(TestBlah.java:38)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 41 common frames omitted
00:03:10.677 [main] INFO  d.f.e.p.runtime.AbstractProcess - construct AbstractPostgresConfig{storage=Storage{dbDir=C:\Users\Jewel\AppData\Local\Temp\postgresql-embed-e5700658-a2b2-47c5-931b-104d1dd801fd\db-content-ea475991-1b59-4860-9182-7da07eff2627, dbName='postgres', isTmpDir=true}, network=Net{host='localhost', port=49577}, timeout=Timeout{startupTimeout=15000}, credentials=Credentials{username='postgres', password='postgres'}, args=[], additionalInitDbParams=[-E, SQL_ASCII, --locale=C, --lc-collate=C, --lc-ctype=C]} 
00:03:10.681 [main] WARN  d.f.e.p.runtime.AbstractProcess - Could not delete pid file: C:\a-tmp\pgsql-10.1-1\pgsql\bin\initdb.pid 
00:03:10.681 [main] WARN  r.y.q.e.postgresql.PostgresProcess - Failed to run command initdb 
java.io.IOException: Cannot run program "runas /trustlevel:0x20000": CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at de.flapdoodle.embed.process.runtime.ProcessControl.start(ProcessControl.java:205)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:98)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.InitDbProcess.<init>(InitDbProcess.java:56)
	at ru.yandex.qatools.embed.postgresql.InitDbExecutable.start(InitDbExecutable.java:23)
	at ru.yandex.qatools.embed.postgresql.InitDbExecutable.start(InitDbExecutable.java:14)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.runCmd(PostgresProcess.java:119)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.runCmd(PostgresProcess.java:74)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.onBeforeProcess(PostgresProcess.java:196)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:82)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.<init>(PostgresProcess.java:68)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:25)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:13)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:140)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:113)
	at Test.unit.TestBlah.blah(TestBlah.java:38)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 41 common frames omitted
00:03:10.684 [main] ERROR d.f.e.p.runtime.AbstractProcess - failed to call start() 
java.io.IOException: Cannot run program "runas /trustlevel:0x20000": CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at de.flapdoodle.embed.process.runtime.ProcessControl.start(ProcessControl.java:205)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:98)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.<init>(PostgresProcess.java:68)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:25)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:13)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:140)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:113)
	at Test.unit.TestBlah.blah(TestBlah.java:38)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 32 common frames omitted
00:03:10.684 [main] INFO  d.f.e.p.runtime.AbstractProcess - construct AbstractPostgresConfig{storage=Storage{dbDir=C:\Users\Jewel\AppData\Local\Temp\postgresql-embed-e5700658-a2b2-47c5-931b-104d1dd801fd\db-content-ea475991-1b59-4860-9182-7da07eff2627, dbName='postgres', isTmpDir=true}, network=Net{host='localhost', port=49577}, timeout=Timeout{startupTimeout=15000}, credentials=Credentials{username='postgres', password='postgres'}, args=[], additionalInitDbParams=[-E, SQL_ASCII, --locale=C, --lc-collate=C, --lc-ctype=C]} 

java.lang.NullPointerException
	at de.flapdoodle.embed.process.runtime.AbstractProcess.getProcessId(AbstractProcess.java:232)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.isProcessRunning(AbstractProcess.java:225)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.stopInternal(PostgresProcess.java:145)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.stop(AbstractProcess.java:170)
	at de.flapdoodle.embed.process.runtime.AbstractProcess.<init>(AbstractProcess.java:119)
	at ru.yandex.qatools.embed.postgresql.AbstractPGProcess.<init>(AbstractPGProcess.java:19)
	at ru.yandex.qatools.embed.postgresql.PostgresProcess.<init>(PostgresProcess.java:68)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:25)
	at ru.yandex.qatools.embed.postgresql.PostgresExecutable.start(PostgresExecutable.java:13)
	at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:101)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:140)
	at ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.start(EmbeddedPostgres.java:113)
	at Test.unit.TestBlah.blah(TestBlah.java:38)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)


Process finished with exit code -1
```

--- a/src/main/java/ru/yandex/qatools/embed/postgresql/EmbeddedPostgres.java
+++ b/src/main/java/ru/yandex/qatools/embed/postgresql/EmbeddedPostgres.java
@@ -2,7 +2,9 @@ package ru.yandex.qatools.embed.postgresql;
 
 import de.flapdoodle.embed.process.config.IRuntimeConfig;
 import de.flapdoodle.embed.process.distribution.IVersion;
+import de.flapdoodle.embed.process.distribution.Platform;
 import de.flapdoodle.embed.process.io.directories.FixedPath;
+import de.flapdoodle.embed.process.runtime.ICommandLinePostProcessor;
 import de.flapdoodle.embed.process.store.PostgresArtifactStoreBuilder;
 import ru.yandex.qatools.embed.postgresql.config.AbstractPostgresConfig;
 import ru.yandex.qatools.embed.postgresql.config.PostgresDownloadConfigBuilder;
@@ -11,6 +13,7 @@ import ru.yandex.qatools.embed.postgresql.config.RuntimeConfigBuilder;
 
 import java.io.IOException;
 import java.nio.file.Path;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Optional;
 
@@ -68,7 +71,36 @@ public class EmbeddedPostgres {
                         .download(new PostgresDownloadConfigBuilder()
                                 .defaultsForCommand(Command.Postgres)
                                 .build()))
+                .commandLinePostProcessor(privilegedWindowsRunasPostprocessor())
                 .build();
+    }
+
+    private static ICommandLinePostProcessor privilegedWindowsRunasPostprocessor() {
+        if (Platform.detect().equals(Platform.Windows)) {
+            try {
+                // Based on https://stackoverflow.com/a/11995662
+                final int adminCommandResult = Runtime.getRuntime().exec("net session").waitFor();
+                if (adminCommandResult == 0) {
+                    return runWithoutPrivileges();
+                }
+            } catch (Exception e) {
+                // Log maybe?
+            }
+        }
+        return doNothing();
+    }
+
+    private static ICommandLinePostProcessor runWithoutPrivileges() {
+        return (distribution, args) -> {
+            if (args.size() > 0 && args.get(0).endsWith("postgres.exe")) {
+                return Arrays.asList("runas", "/trustlevel:0x20000", String.format("\"%s\"", String.join(" ", args)));
+            }
+            return args;
+        };
+    }
+
+    private static ICommandLinePostProcessor doNothing() {
+        return (distribution, args) -> args;
     }
 
     /**
@@ -90,6 +122,7 @@ public class EmbeddedPostgres {
                                 .defaultsForCommand(cmd)
                                 .packageResolver(new PackagePaths(cmd, cachedDir))
                                 .build()))
+                .commandLinePostProcessor(privilegedWindowsRunasPostprocessor())
                 .build();
     }
 

